### PR TITLE
DCOS-6523: Update package configuration

### PR DESCRIPTION
**Pull request by jfurrow**

### CODE REVIEW

**Comment by mesosphere-web** [comment]
Reviewer: @mlunoe

**Comment by jfurrow** [comment]
I forgot to include the `plan` component updating stuff, so closing this for now.

File: src/js/components/ServiceActions.js [hunk 1]
```diff
@@ -0,0 +1,117 @@
...
+import InternalStorageMixin from '../mixins/InternalStorageMixin';
+import Service from '../structs/Service';
+import UpdateConfigModal from './modals/UpdateConfigModal';
+
+const METHODS_TO_BIND = [
+  'handleEditConfigClick',
+  'handleConfigModalClose'
+];
+
+class ServiceActions extends mixin(InternalStorageMixin, StoreMixin) {
```
Comment (kennyt) [inline src/js/components/ServiceActions.js]: `ServiceActions` sounds like an `Actions` file (those things that make requests)

Comment (jfurrow) [inline src/js/components/ServiceActions.js]: That's a good point, I'll figure out something else for this component.

File: src/js/components/ServiceActions.js [hunk 2]
```diff
@@ -0,0 +1,117 @@
...
+  handleConfigModalClose() {
+    this.setState({editConfigModalOpen: false});
+  }
+
+  render() {
+    let {service} = this.props;
+    let {editConfigModalOpen} = this.state;
+    let {servicePackage} = this.internalStorage_get();
+
+    let editButotnClasses = classNames('button button-inverse button-stroke', {
```
Comment (kennyt) [inline src/js/components/ServiceActions.js]: `Button` is spelled wrong

File: src/js/components/modals/UpdateConfigModal.js [hunk 3]
```diff
@@ -0,0 +1,249 @@
...
+    }];
+
+    METHODS_TO_BIND.forEach((method) => {
+      this[method] = this[method].bind(this);
+    });
+  }
+
+  onCosmosPackagesStoreInstallSuccess() {
+    this.internalStorage_update({
+      updateError: null,
```
Comment (kennyt) [inline src/js/components/modals/UpdateConfigModal.js]: i cannot for the life of me see where `updateError` ever gets set to anything other than `null`

Comment (jfurrow) [inline src/js/components/modals/UpdateConfigModal.js]: Yeah, I had some rebase issues so I closed this PR yesterday. I am now cherry-picking in some of the upgrade progress detail modal stuff right now and will have this back up ASAP :)

File: src/js/components/modals/UpdateConfigModal.js [hunk 4]
```diff
@@ -0,0 +1,249 @@
...
+  'handleModalClose',
+  'getConfigSubmit'
+];
+
+class UpdateConfigModal extends mixin(InternalStorageMixin, StoreMixin) {
+  constructor() {
+    super(...arguments);
+
+    this.internalStorage_set({
+      hasFormErrors: false,
```
Comment (kennyt) [inline src/js/components/modals/UpdateConfigModal.js]: why are these in the `internalStorage`?

i've always been curious about why we've had this hatred vs state (talking about our general application, not just this component). apparently, `state` is for things that change over time. these all fit that description. so why not just use what `React` gives us?

having to `forceUpdate` not only ignores `shouldComponentUpdate`, it's also error prone. plus, you have to keep typing in `internalStorage_get` / `internalStorage_set`

**Comment by kennyt** [comment]
just noticed this was closed 19 hours ago. sorry about that, github showed the PR as open

**Pull request closed without merging**